### DISCUSSION TITLE
data button response

### DISCUSSION THREAD

**Discussion by jwitvoet** [jspsych/jsPsych#267] [Q&A]
Hi Josh,

Is is possible to record the data of all button presses when using the button-response plugin? 

My stimulus is a video and participants have to respond several times throughout the video using 2 buttons. I thus need the data of every button that was pressed during the trial. 

Is there any way to accomplish this?

**Comment by jodeleeuw** [comment]
You would need to modify the plugin file to do this. Right now it is set up to be a single button response.

The key line is [L68 of the button response plugin](https://github.com/jodeleeuw/jsPsych/blob/master/plugins/jspsych-button-response.js#L68). That line causes the trial to end when a button is pressed. You could modify the logic inside the after_response function to record the button press and the timing.

Then you could end the trial when the video file ends, borrowing the technique from [the video plugin](https://github.com/jodeleeuw/jsPsych/blob/master/plugins/jspsych-video.js):

``` javascript
document.getElementById('jspsych-video-player').onended = function(){
  end_trial();
}
```

If you are comfortable modifying the video plugin to include this as an option, it would be a great addition to the plugin and I would be eager to add it to the library.

**Comment by jwitvoet** [comment]
In the button-response plugin, I've set response_ends_trial to false, and i've set timing_response to the end of the video (manually). I've tried to use an event handler to record the button press but this didn't work. Any idea's on how I could record each button press in L68 of the plugin?

**Comment by jodeleeuw** [comment]
I would suggest modifying the after_response function. Perhaps create an array to hold the responses, and then push each response onto the array. When the trial ends, you can record the JSON encoding of that array as data for the trial.

Sorry that's a bit vague, but hopefully it gets you pointed in the right direction.

**Comment by jodeleeuw** [comment]
I'm closing this issue, but feel free to continue to use the thread if you have further questions.

**Comment by jwitvoet** [comment]
With help we were able to change the button-response plugin to record every button press and rt. 

I've attached the plugin. Maybe it'll be useful to someone else. 
[jspsych-button-response kopie.js.zip](https://github.com/jodeleeuw/jsPsych/files/325327/jspsych-button-response.kopie.js.zip)

**Comment by ceciliemoeller** [comment]
Hi Josh,
I found this four year old thread when searching for a way to record measures of beats per minute (BPM), i.e. several button presses within the same 15 sec audio trial (later I will need video trials as well). I am fairly confident that I will be able to use the modified plugin above (thanks for posting this, jwitvoet) to convince the current audio-button-response plugin to do the trick. I have more confidence than experience, though, and it will take me a while, so before I do so, I would like to ask if you know of anything created more recently that can be used to measure tapping data? 
Thanks!
Cecilie

**Comment by jodeleeuw** [comment]
Hi Cecilie,

I don't think there are any more recent developments that will support this. I know @davidjohnbaker1 was doing some work with music and jsPsych. @davidjohnbaker1 do you know of anything that fits this task?

Josh

**Comment by ceciliemoeller** [comment]
Thanks Josh,
this was actually easier than I thought. I've essentially copied the relevant lines from the script above and merged them into the audio-button-response plugin and it works fine. I am going to see if I can make it log the button presses instead of releases (for precision) and then I can post the plugin here, or would you like med to try to make a proper pull request? 
/Cecilie
